File: 1.java/ACE/ace Desktop/Ace Desktop/src/Entite/Evenement.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Entite;
import java.sql.Date;
//mport java.util.Date;
/**
 *
 * @author farah
 * 
 */
public class Evenement {
  
     private int id_event;
    private float duree;
    private float prix;
    private int id;
    String nom_event;
   Date date_deb;
   Date date_fin;

    public Evenement(int id_event, float duree, float prix, int id, String nom_event, Date date_deb, Date date_fin) {
        this.id_event = id_event;
        this.duree = duree;
        this.prix = prix;
        this.id = id;
        this.nom_event = nom_event;
        this.date_deb = date_deb;
        this.date_fin = date_fin;
    }

    public Evenement(float duree, float prix, int id, String nom_event, Date date_deb, Date date_fin) {
        this.duree = duree;
        this.prix = prix;
        this.id = id;
        this.nom_event = nom_event;
        this.date_deb = date_deb;
        this.date_fin = date_fin;
    }

    public Evenement(float duree, float prix, int id, Date date_deb, Date date_fin) {
        this.duree = duree;
        this.prix = prix;
        this.id = id;
        this.date_deb = date_deb;
        this.date_fin = date_fin;
    }

  
    

   
    
      

    public Evenement() {
    }

    public Evenement(float duree, float prix) {
        this.duree = duree;
        this.prix = prix;
    }

    public int getId_event() {
        return id_event;
    }

    public void setId_event(int id_event) {
        this.id_event = id_event;
    }

    public float getDuree() {
        return duree;
    }

    public void setDuree(float duree) {
        this.duree = duree;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom_event() {
        return nom_event;
    }

    public void setNom_event(String nom__event) {
        this.nom_event = nom_event;
    }

    public Date getDate_deb() {
        return date_deb;
    }

    public void setDate_deb(Date date_deb) {
        this.date_deb = date_deb;
    }

    public Date getDate_fin() {
        return date_fin;
    }

    public void setDate_fin(Date date_fin) {
        this.date_fin = date_fin;
    }

    @Override
    public String toString() {
        return   id_event + "\t         " + duree +"\t             "+ prix + "\t             " + id + "\t                   "+ nom_event +"\t                 " + date_deb + "\t              " + date_fin ;
    }

    
  


   
    
 

   
}

File: Integration/src/Entity/Evenement.php
<?php

namespace App\Entity;

use App\Repository\EvenementRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
use Symfony\Component\Serializer\Serializer;
use Symfony\Component\Serializer\SerializerInterface;
use Symfony\Component\Serializer\Annotation\Groups;


#[ORM\Entity(repositoryClass: EvenementRepository::class)]
/**
     * @UniqueEntity(fields={"nom_event"}, message="Ce nom d'événement est déjà utilisé.")
     */
class Evenement
{

    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id_event = null;

    #[ORM\Column]
    
    #[Assert\NotBlank(message:"veuillez remplir ce champ de la duree")]
    #[Assert\Regex(
        pattern: "/^[0-9]+$/",
        message: "This value is not valid."
    )]
    #[Groups("event")]
    private ?float $duree = null;

    #[ORM\Column]
    #[Assert\NotBlank(message:"veuillez remplir ce champ du prix")]
    #[Assert\Regex(
        pattern: "/^[0-9]+$/",
        message: "This value is not valid."
    )]
    #[Groups("event")]
    private ?float $prix = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\NotBlank(message:"veuillez remplir ce champ de la date de debut")]
    #[Groups("event")]
    private ?\DateTimeInterface $date_deb = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
#[Assert\NotBlank(message: "Veuillez remplir ce champ de la date de fin")]
#[Assert\NotNull(message: "Veuillez remplir ce champ de la date de début")]
#[Assert\GreaterThan(propertyPath: "date_deb", message: "La date de fin doit être après la date de début")]
#[Groups("event")]

private ?\DateTimeInterface $date_fin = null;
    

    #[ORM\Column(length: 25)]
    #[Assert\NotBlank(message:"veuillez remplir ce champ du nom de levent")]

    #[Assert\Regex(
        pattern:"/^[^0-9]+$/",
        message:"Le nom ne doit pas contenir de chiffres"
    )]
    #[Groups("event")]

    private ?string $nom_event = null;
   

   

    #[ORM\OneToMany(mappedBy: 'evenements', targetEntity: Publicite::class)]
    
    private Collection $publicites;

    #[ORM\ManyToOne(inversedBy: 'evenements')]
    #[Assert\NotBlank(message:"veuillez remplir ce champs SVP")]

    private ?Destination $destin = null;

    public function __construct()
    {
        $this->publicites = new ArrayCollection();
    }

   

    public function getId(): ?int
    {
        return $this->id_event;
    }

    public function getDuree(): ?float
    {
        return $this->duree;
    }

    public function setDuree(float $duree): self
    {
        if ($duree <= 0) {
            throw new \InvalidArgumentException('La durée doit être supérieure à 0');
        }
        $this->duree = $duree;
        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        if ($prix < 0) {
            throw new \InvalidArgumentException('Le prix doit être un nombre positif');
        }
        $this->prix = $prix;
        return $this;
    }

    public function getDateDeb(): ?\DateTimeInterface
    {
        return $this->date_deb;
    }

    public function setDateDeb(\DateTimeInterface $date_deb): self
    {
        if (empty($date_deb)) {
            throw new \InvalidArgumentException('La date de début ne peut pas être vide');
        }
        $this->date_deb = $date_deb;
        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->date_fin;
    }

    public function setDateFin(\DateTimeInterface $date_fin): self
    {
        if (empty($date_fin)) {
            throw new \InvalidArgumentException('La date de fin ne peut pas être vide');
        }
        $this->date_fin = $date_fin;
        return $this;
    }

    public function getNomEvent(): ?string
    {
        return $this->nom_event;
    }

    public function setNomEvent(string $nom_event): self
    {
        if (empty(trim($nom_event))) {
            throw new \InvalidArgumentException('Le nom de l\'événement ne peut pas être vide');
        }
        $this->nom_event = $nom_event;
        
        return $this;
    }

  

    /**
     * @return Collection<int, Publicite>
     */
    public function getPublicites(): Collection
    {
        return $this->publicites;
    }

    public function addPublicite(Publicite $publicite): self
    {
        if (!$this->publicites->contains($publicite)) {
            $this->publicites->add($publicite);
            $publicite->setEvenements($this);
        }

        return $this;
    }

    public function removePublicite(Publicite $publicite): self
    {
        if ($this->publicites->removeElement($publicite)) {
            // set the owning side to null (unless already changed)
            if ($publicite->getEvenements() === $this) {
                $publicite->setEvenements(null);
            }
        }

        return $this;
    }

    public function __toString(): string
    {
        return $this->nom_event ?? '';
    }

    public function getDestin(): ?Destination
    {
        return $this->destin;
    }

    public function setDestin(?Destination $destin): self
    {
        

        if (empty($destin)) {
            throw new \InvalidArgumentException('La destination ne peut pas être vide');
        }
        $this->destin = $destin;
        return $this;
    }

  
   

   
   

   
}
